stringdtype/stringdtype/src/dtype.c

```diff
@@ -277,6 +278,15 @@ static PyMethodDef StringDType_methods[] = {
         {NULL},
 };
 
+// Implementation of PyArray_CompareFunc.
+int
+compare_strings(char **a, char **b, PyArrayObject *arr)
+{
+    ss *ss_a = (ss *)*a;
+    ss *ss_b = (ss *)*b;
+    return strcmp(ss_a->buf, ss_b->buf);
```

Comment:
Rather than using `strcmp`, first check if the lengths are identical and then if they are, use the length to do `strncmp`.

Reply:
Oops, discard that suggestion, it only makes sense for equality comparison, not for generic comparisons.